### PR TITLE
feature: Initial support for SemanticDB in bazel projects

### PR DESCRIPTION
**Pull request by jkciesluk**

### CODE REVIEW

File: metals/src/main/scala/scala/meta/internal/metals/MetalsLspService.scala [hunk 1]
```diff
@@ -1284,7 +1284,7 @@ class MetalsLspService(
       // reliable way to await until those notifications appear.
       for {
         targetroot <- buildTargets.targetRoots(report.getTarget)
```
Comment (tgodzik) [inline metals/src/main/scala/scala/meta/internal/metals/MetalsLspService.scala:1286]: Maybe we should do `toTargetRoot` inside `buildTarget.targetRoot` method? Otherwise we might forget to put it in one place

File: metals/src/main/scala/scala/meta/internal/metals/MetalsEnrichments.scala [hunk 2]
```diff
@@ -533,6 +533,17 @@ object MetalsEnrichments
       filename.endsWith(".jar") || filename.endsWith(".srcjar")
     }
 
+    def toTargetRoot: AbsolutePath = {
+      if (path.isJar) {
+        val filename = path.toNIO.getFileName.toString
+        val filename0 = filename.stripSuffix(".jar").stripSuffix(".srcjar")
+        val targetroot = path.parent.resolve(s"_semanticdb/$filename0")
```
Comment (tgodzik) [inline metals/src/main/scala/scala/meta/internal/metals/MetalsEnrichments.scala:540]: How does the directory structure look like in case of Bazel? `_semanticdb` seems like it could be fragile.

Comment (jkciesluk) [reply metals/src/main/scala/scala/meta/internal/metals/MetalsEnrichments.scala:540]: Based on simple projects I've seen, the `_semanticdb` directory gets created in the same place as the targetroot jar.

Comment (tgodzik) [reply metals/src/main/scala/scala/meta/internal/metals/MetalsEnrichments.scala:540]: Let's add a comment that it's used for Bazel.

File: metals/src/main/scala/scala/meta/internal/metals/MetalsEnrichments.scala [hunk 3]
```diff
@@ -533,6 +533,17 @@ object MetalsEnrichments
       filename.endsWith(".jar") || filename.endsWith(".srcjar")
     }
 
+    def toTargetRoot: AbsolutePath = {
```
Comment (tgodzik) [inline metals/src/main/scala/scala/meta/internal/metals/MetalsEnrichments.scala:536]:
```suggestion
    def resolveIfJar: AbsolutePath = {
```
maybe?

File: metals/src/main/scala/scala/meta/internal/metals/watcher/FileWatcher.scala [hunk 4]
```diff
@@ -111,7 +110,6 @@ object FileWatcher {
     buildTargets.sourceItems.foreach(collect)
 
     buildTargets.allTargetRoots
-      .filterNot(_.isJar)
```
Comment (tgodzik) [inline metals/src/main/scala/scala/meta/internal/metals/watcher/FileWatcher.scala:114]: This jar target root will be the initial one which we resolve in `MetalsEntichments` ? If so should we resolve it here?

File: tests/slow/src/test/scala/tests/bazel/BazelLspSuite.scala [hunk 5]
```diff
@@ -161,17 +161,17 @@ class BazelLspSuite
...
            |Tags
-           |  library
+           |  application
            |
            |Languages
            |  scala
            |
            |Capabilities
            |  Debug <- NOT SUPPORTED
-           |  Run <- NOT SUPPORTED
```
Comment (tgodzik) [inline tests/slow/src/test/scala/tests/bazel/BazelLspSuite.scala:174]: Why the change? Does Bazel supports run?

Comment (jkciesluk) [reply tests/slow/src/test/scala/tests/bazel/BazelLspSuite.scala:174]: Previously the "first" target was a library, now its a binary

**Review by tgodzik — APPROVED** [review]
LGTM!
